### PR TITLE
feat: Drop support for iOS 13,14 WPB-4710

### PR DESCRIPTION
**Pull request by KaterinaWire**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-4710" title="WPB-4710" target="_blank">StoryWPB-4710</a>  Drop support for iOS 13,14
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

 Drop support for iOS 13,14.
 There is an issue with that: `kUTTypeJPEG` and other types are deprecated in iOS 15.0 and should use UTType instead.

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Review by David-Henner — APPROVED** [review]

File: wire-ios-data-model/Source/Model/Conversation/ZMConversation+Participants.swift [hunk 1]
```diff
@@ -258,14 +258,14 @@ extension ZMConversation {
                 do {
                     try await mlsService.addMembersToConversation(with: mlsUsers, for: mlsGroupID)
 
-                    context.perform {
+                    await context.perform {
```
Comment (netbe) [inline wire-ios-data-model/Source/Model/Conversation/ZMConversation+Participants.swift:261]: why?

File: wire-ios-data-model/Source/Model/Conversation/ZMConversation+Participants.swift [hunk 2]
```diff
@@ -258,14 +258,14 @@ extension ZMConversation {
...
                         completion(.success(()))
                     }
 
                 } catch {
                     Logging.mls.error("failed to add members to conversation (\(String(describing: qualifiedID))): \(String(describing: error))")
 
-                    context.perform {
+                    await context.perform {
```
Comment (netbe) [inline wire-ios-data-model/Source/Model/Conversation/ZMConversation+Participants.swift:268]: not sure to understand the need here?

File: wire-ios-notification-engine/WireNotificationEngine.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -747,7 +743,6 @@
 				CURRENT_PROJECT_VERSION = 1;
 				DEVELOPMENT_TEAM = EDF3JCE8BC;
 				GENERATE_INFOPLIST_FILE = YES;
-				IPHONEOS_DEPLOYMENT_TARGET = 15.2;
```
Comment (netbe) [inline wire-ios-notification-engine/WireNotificationEngine.xcodeproj/project.pbxproj:750]: any issue in removing this?

File: wire-ios-utilities/Source/UTIHelper.swift [hunk 4]
```diff
@@ -224,54 +192,44 @@ public final class UTIHelper: NSObject {
...
-            }
+        var mimeType: String?
+        if let utType = utType {
+            mimeType = mime(from: utType)
+        }
 
-            return mimeType
-        } else {
-            return iOS13ConvertToMime(fileExtension: fileExtension)
+        /// HACK: when resolving .pkpass file extension, the above method returns nil, fallback to iOS 13- method.
```
Comment (netbe) [inline wire-ios-utilities/Source/UTIHelper.swift]: is it still needed?

File: wire-ios-utilities/Source/UTIHelper.swift [hunk 5]
```diff
@@ -224,54 +192,44 @@ public final class UTIHelper: NSObject {
...
+        if utType == nil {
+            utType = UTType(tag: fileExtension, tagClass: .filenameExtension, conformingTo: nil)
+        }
 
-            var mimeType: String?
-            if let utType = utType {
-                mimeType = mime(from: utType)
-            }
+        #if targetEnvironment(simulator)
+        /// HACK: hard code MIME when preferredMIMEType is nil for M1 simulator, we should file a ticket to apple for this issue
```
Comment (netbe) [inline wire-ios-utilities/Source/UTIHelper.swift]: is it really an issue for M1 simulator ? I found https://developer.apple.com/forums/thread/685968

File: wire-ios-utilities/Source/UTIHelper.swift [hunk 6]
```diff
@@ -306,22 +264,12 @@ public final class UTIHelper: NSObject {
...
-            return mime(from: utType)
-        } else {
-            let unmanagedMime = UTTypeCopyPreferredTagWithClass(uti as CFString, kUTTagClassMIMEType)
-
-            guard let retainedValue = unmanagedMime?.takeRetainedValue() else {
-                return nil
-            }
-
-            return retainedValue as String
+        print("something")
```
Comment (netbe) [inline wire-ios-utilities/Source/UTIHelper.swift]: remove print

File: wire-ios/Wire-iOS/Sources/UserInterface/Components/Controllers/SplitViewController.swift [hunk 7]
```diff
@@ -322,11 +322,7 @@ final class SplitViewController: UIViewController, SplitLayoutObservable {
     }
 
     private var isiOSAppOnMac: Bool {
-        if #available(iOS 14.0, *) {
-            return ProcessInfo.processInfo.isiOSAppOnMac
-        }
-
-        return false
+        return ProcessInfo.processInfo.isiOSAppOnMac
```
Comment (netbe) [inline wire-ios/Wire-iOS/Sources/UserInterface/Components/Controllers/SplitViewController.swift:325]: did not know we had this check :)

File: wire-ios-utilities/Source/UTIHelper.swift (hunk 6 above)
Comment (KaterinaWire) [reply wire-ios-utilities/Source/UTIHelper.swift]: done

File: wire-ios-data-model/Source/Model/Conversation/ZMConversation+Participants.swift (hunk 1 above)
Comment (KaterinaWire) [reply wire-ios-data-model/Source/Model/Conversation/ZMConversation+Participants.swift:261]: `perform` is async method for iOS 15 and therefore `await` is required

```
@available(macOS 12.0, iOS 15.0, tvOS 15.0, watchOS 8.0, *)
extension NSManagedObjectContext {

    public func performAndWait<T>(_ block: () throws -> T) rethrows -> T

    public func perform<T>(schedule: NSManagedObjectContext.ScheduledTaskType = .immediate, _ block: @escaping () throws -> T) async rethrows -> T
}
```

File: wire-ios-data-model/Source/Model/Conversation/ZMConversation+Participants.swift (hunk 2 above)
Comment (KaterinaWire) [reply wire-ios-data-model/Source/Model/Conversation/ZMConversation+Participants.swift:268]: same reply as previous

File: wire-ios-notification-engine/WireNotificationEngine.xcodeproj/project.pbxproj (hunk 3 above)
Comment (KaterinaWire) [reply wire-ios-notification-engine/WireNotificationEngine.xcodeproj/project.pbxproj:750]: no, it was for test targets

File: wire-ios-utilities/Source/UTIHelper.swift (hunk 4 above)
Comment (KaterinaWire) [reply wire-ios-utilities/Source/UTIHelper.swift]: no, removed

File: wire-ios-utilities/Source/UTIHelper.swift (hunk 5 above)
Comment (KaterinaWire) [reply wire-ios-utilities/Source/UTIHelper.swift]: removed

**Review by netbe — APPROVED** [review]

**Pull request merged**